doc: Handle READTHEDOCS_VERSION in preview (#12158)
cilium-team/cilium: Documentation/contributing/testing/documentation.rst

This generates documentation files and starts a web server using a Docker container. You can
view the updated documentation by opening either ``Documentation/_build/html/index.html`` or
http://localhost:9081 in a browser.

.. note::

>    By default, ``render-docs-live-preview`` generates a preview assuming it's for the latest
>    version (i.e. from the HEAD of the master branch that has not been released) regardless of
>    which Cilium branch you are in. You can target a specific branch by specifying ``READTHEDOCS_VERSION``
>    environment variable:
> 

Reviewer: I found this a bit confusing with respect to the branches. Would the following be just as clear to you?

Suggestion:

```suggestion
   By default, ``render-docs-live-preview`` generates a preview with instructions to install Cilium from the latest
   version on GitHub (i.e. from the HEAD of the master branch that has not been released) regardless of
   which Cilium branch you are in. You can target a specific branch by specifying ``READTHEDOCS_VERSION``
   environment variable:

```

Author: no. your suggestion is clearer :). updated.
